packages/amplify-cli/src/init-steps/preInitSetup.ts

```diff
@@ -3,11 +3,24 @@ import { execSync } from 'child_process';
 import * as fs from 'fs-extra';
 import * as url from 'url';
 import { generateLocalEnvInfoFile } from './s9-onSuccess';
+import { printer, prompter } from '@aws-amplify/amplify-prompts';
+import { isNewProject } from './s0-analyzeProject';
+
+export const getPreInitSetup = (recommendGen2: boolean) => {
+  if (recommendGen2) {
+    return preInitSetupBasic;
+  } else {
+    return async (context) => {
+      await gen2Recommendation(context);
+      await preInitSetupBasic(context);
+    };
+  }
+};
 
 /**
  * Executes before init
  */
-export const preInitSetup = async (context: $TSContext): Promise<$TSContext> => {
+export const preInitSetupBasic = async (context: $TSContext): Promise<$TSContext> => {
```

Comment:
I think it's fine to leave the name as-is. And possible it doesn't need to be exported anymore.
```suggestion
export const preInitSetup = async (context: $TSContext): Promise<$TSContext> => {
```

Reply:
Done 1e8e25ba3bf8efb06a0096f000c74f7d1dc4caff